### DISCUSSION TITLE
How to change the line spacing of free text annotation?

### DISCUSSION THREAD

**Discussion by Fishdozer** [sumatrapdfreader/sumatrapdf#3907] [Q&A]
The line spacing of my annotaition is too wide. When I add a new line in annotations, a blank line automatically appears. You can see it in my screenshot. Maybe some misoperation leads to this situation. I want to reset the linewidth as default, but I didn't find a solution for this. I tried each option in "annotations" window and reloading the program, but they wouldn't help. Please.
![231124](https://github.com/sumatrapdfreader/sumatrapdf/assets/108393437/5a0e10bb-1397-498c-be6e-e0750bc6bed4)

**Comment by GitHubRulesOK** [comment]
PDF is media graphics so unlike HTML/DOC does not use inline feeds in text nor reflow.

Annotation is an odd beast it has NO definable line spacing, open the comment in any reader and it adjusts.
but unlike other pdf constructs "Free Text" can temporarily reflow and have variable line feeds, and often be inclusive of HTML style contents.

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/e6552623-f42c-4ba0-9300-1fdebd98307c)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/81cd0271-a045-4cbc-8413-f68773e8ad74)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/9bf57c9f-91cf-4a73-8194-ea07ee3ba9c4)


Another odd thing about FreeText is it is stored in many formats in the one file so for above example here are snips from the ONE result after Acrobat adjusted widths. It has decided on this occasion to hold the block as 
- 1 full string  as like HTML with no starting tags but ends with `</p></body>` ! so a reader could determine the whole page is one reflow paragraph
- many partial lines from the source layout (seen as 12 point high at -14.400001 line spacing)
- all words as individuals (12 points high at -14.4 line spacing)

```
7 0 obj
<</AP<</N 13 0 R>>/BS<</Type/Border/W 0>>/Contents(Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.)/DA(/Helv 12 Tf 0 0 0 rg)/DS(font: Helvetica,sans-serif 12.0pt; text-align:left; color:#000000 )/F 4/M(D:20231124102328Z)/NM(5931eb59-039b-4a0e-a6ae-0cf954bb862f)/P 3 0 R/RC(<?xml version="1.0"?><body xmlns="http://www.w3.org/1999/xhtml" xmlns:xfa="http://www.xfa.org/schema/xfa-data/1.0/" xfa:APIVersion="Acrobat:23.6.0" xfa:spec="2.0.2"  style="font-size:12.0pt;text-align:left;color:#000000;font-weight:normal;font-style:normal;font-family:Helvetica,sans-serif;font-stretch:normal"><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p></body>)/Rect[415.369 490.875 822.858 590.875]/Subtype/FreeText/T(lez)/Type/Annot>>
endobj

9 0 obj
<</Type/XObject/Subtype/Form/BBox[0 0 200 99.99997]/Matrix[1 0 0 1 0 0]/Resources<</Font<</Helv 8 0 R>>>>/Length 945>>
stream
0 w
0 0 200 99.99997 re
W
n
BT
0 0 0 rg
0 104.79997 Td
0 -14.400001 Td
/Helv 12 Tf
(Lorem ipsum dolor sit amet, ) Tj
0 -14.400001 Td
/Helv 12 Tf
(consectetur adipiscing elit, sed do ) Tj
0 -14.400001 Td
/Helv 12 Tf
(eiusmod tempor incididunt ut labore ) Tj
0 -14.400001 Td
....
/Helv 12 Tf
(sint occaecat cupidatat non proident, ) Tj
0 -14.400001 Td
/Helv 12 Tf
(sunt in culpa qui officia deserunt ) Tj
0 -14.400001 Td
/Helv 12 Tf
(mollit anim id est laborum.) Tj
ET

and from object 13
BT
/Helv 12 Tf
0 g
415.369 581.4 Td
(Lorem ) Tj
37.348 0 Td
(ipsum ) Tj
35.343 0 Td
(dolor ) Tj
30.018 0 Td
(sit ) Tj
15.334 0 Td
(amet, ) Tj
33.346 0 Td
(consectetur ) Tj
...                                            in total about 150 lines of text positions !!
29.344 0 Td
(id ) Tj
12.674 0 Td
(est ) Tj
19.341 0 Td
(laborum.) Tj
ET
Q
endstream

```

**Reply by GitHubRulesOK** [reply]
So back to the Question How do other editors produce the desired result from a string when there are no line feeds ?
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/759365fc-16d9-4233-96a0-f194a034ba2e)

And the answer is they are more complex so Acrobat Reader has much code from Acrobat Pro included in its abilities, and other editors can adjust the notional line spacing. SumatraPDF based on MuPDF can use those setting for rendering but does not (yet ?) use multiple editor settings

Here is your inputs from another editor source and the whole structure is different, imbedded carriage returns and notional 115% spacing in a more compact 2 areas rather than Acrobats 3.
```
11 0 obj
<</AP<</N 14 0 R>>/BS<</W 0>>/Contents(a line\rb line\rc line)/DA(0 0 0 rg /Helv 12 Tf)/DS(line-height:13.8pt;line-spacing:115%;font-family:Helvetica,sans-serif;font-size:12pt;color:#000000;)/F 4/IT/FreeText/M(D:20231124112917Z)/NM(5931eb59-039b-4a0e-a6ae-0cf954bb862f)/P 10 0 R/Rect[415.369 490.875 822.858 590.875]/Subtype/FreeText/T(lez)/Type/Annot>>
endobj

part of object 14
BT
0 g 0 Tc 0 Tw 100 Tz 0 Tr/F0 12 Tf 415.369 580.015 Td
(a line)Tj
0 -15.8106 Td
(b line)Tj
0 -15.8106 Td
(c line)Tj
ET

```

**Reply by Fishdozer** [reply]
I use one of other PDF editors ("Foxit") to modify the annotation. Then the line spacing indeed be changed to a standard case and I saved the PDF file, whose annotation is with correct line spacing. But when I re-edit the free text annotation in my Sumatra, huge line spacing appears again. :(

**Reply by GitHubRulesOK** [reply]
there is both an existing open issue  https://github.com/sumatrapdfreader/sumatrapdf/issues/1694
and a related one https://github.com/sumatrapdfreader/sumatrapdf/issues/2873
